### PR TITLE
Migrate domain objects from MasterData project

### PR DESCRIPTION
**Pull request by kft**
## Description
Migrated domain objects including tests from MasterData to MarketRoles solution.
Unused and obsolete objects are removed.

### CODE REVIEW

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/Consumers/Rules/CvrNumberFormatRule.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            return cvrValue.Length == RequiredIdLength;
+        }
+
+        private static bool RangeIsValid(long cvrNumber)
+        {
+            return cvrNumber >= LowerBound && cvrNumber <= UpperBound;
+        }
+
+        private bool IsValidCvrNumber()
```
Comment (erduna) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/Consumers/Rules/CvrNumberFormatRule.cs:46]: "CVR-nummeret er altså bygget op af 8 cifre, og opfylder modulus 11 kontrollen"
https://da.wikipedia.org/wiki/Modulus_11

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/MeteringPoints/AccountingPoint.cs [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            AddBusinessProcess(processId, moveInDate, BusinessProcessType.MoveIn);
+            _consumerRegistrations.Add(new ConsumerRegistration(consumerId, processId));
+            _supplierRegistrations.Add(new SupplierRegistration(energySupplierId, processId));
+        }
+
+        public void EffectuateConsumerMoveIn(ProcessId processId, ISystemDateTimeProvider systemDateTimeProvider)
+        {
+            var businessProcess = GetBusinessProcess(processId, BusinessProcessType.MoveIn);
+
+            businessProcess.Effectuate(systemDateTimeProvider);
```
Comment (erduna) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/MeteringPoints/AccountingPoint.cs:159]: Not sure this is where we have to do it, but currently every effectuation date in messages sent/received is at 22hours zulu time on the day before. Meaning if you change supplier on the 24th, the effectuation date is the 23rd at 22:00:00z

**Review by erduna — DISMISSED** [review]
Any changes based on comments can be done as seperate PR

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/Consumers/CprNumber.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        {
+        }
+
+        public static CprNumber Create(string cprValue)
+        {
+            var formattedValue = cprValue?.Trim();
+
+            if (string.IsNullOrWhiteSpace(formattedValue))
+            {
+                throw new ArgumentException($"'{nameof(cprValue)}' cannot be null or whitespace", nameof(cprValue));
```
[the remark this answers is not in the document]
Comment (kft) [reply source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/Consumers/CprNumber.cs:35]: No. It's a value object for CPR number. And a CPR number cannot be blank :-)

**Review by Korgath — DISMISSED** [review]

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/Consumers/CprNumber.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (erduna) [reply source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/Consumers/CprNumber.cs:35]: What I mean is that in the current DH the cpr can be blank

**Review by erduna — APPROVED** [review]

**Review by Korgath — APPROVED** [review]

**Pull request merged**